Exercise_9/src/MyFramework/app.js

```diff
@@ -23,6 +24,21 @@ class App {
         this.rootElement.innerHTML = content;
       });
   }
+
+  component(name, descriptor) {
+    const customElements = window.document.getElementsByTagName(name);
+    const elementToInsert = parsingService.createTagFromString(descriptor.template);
+
+    for (let index = 0; index < customElements.length; index += 1) {
+      const customElement = customElements.item(index);
+
+      elementToInsert.innerHTML = customElement.innerHTML;
+      descriptor.beforeMount(this, elementToInsert, customElement.attributes);
```

Comment:
I've just realized that we should have common logic for `beforeMount` because for each new component we will do the same stuff except some special logic. I propose to add a common implementation for `beforeMount` after which we should call `descriptor.beforeMount`.